### DISCUSSION TITLE
Update underlying DataSource for a RasterizingTileLayer

### DISCUSSION THREAD

**Discussion by Thoronion** [Mapsui/Mapsui#2444] [Q&A] [answered]
Hi.

We have a RasterizingLayer with an underlying layer. When we want to update the Features of the underlying layer we simply do 

```csharp
Layer _layer = new();
RasterizingLayer _rasterLayer = new(_layer);
...

_layer.DataSource = new IndexedMemoryProvider(NewFeatures());
_rasterLayer.ClearCache();
```

This works well but we want to replace the RasterizingLayer with a RasterizingTileLayer. The code above does not update the map when using RasterizingTileLayer instead. Has anybody been able to update the underlying DataSource of a RasterizingTileLayer?

**Comment by Thoronion** [comment, marked as answer]
After some investigation it seems the problem is with Layer. Replacing Layer with some other ILayer type solves the problem.

**Comment by Thoronion** [comment]
@inforithmics do you know if there is a fix for this in Main comapared to develop/4.1? Maybe it is related to the IndexedMemoryProvider.

**Reply by inforithmics** [reply]
I'll have a look at it.
